lib/fabrik/DirectorManager.js

```diff
@@ -322,36 +322,40 @@ class DirectorManager extends BaseManager {
       .deleteDeploymentProperty(deploymentName, `binding-${id}`);
   }
 
-  diffManifest(deploymentName) {
+  diffManifest(deploymentName, opts) {
     logger.debug(`+-> Checking diff of deployment ${deploymentName}`);
     return this
-      .regenerateManifest(deploymentName)
+      .regenerateManifest(deploymentName, opts)
       .then(manifest => this.director
         .diffDeploymentManifest(deploymentName, manifest)
         .then(result => _.set(result, 'manifest', manifest))
       );
   }
 
-  regenerateManifest(deploymentName) {
+  regenerateManifest(deploymentName, opts) {
     return this.director
       .getDeploymentManifest(deploymentName)
-      .then(manifest => this.generateManifest(deploymentName, {
+      .then(manifest => this.generateManifest(deploymentName, _.extend(opts, {
         previous_manifest: manifest
-      }));
+      })));
   }
 
   generateManifest(deploymentName, opts) {
     const index = opts.network_index || this.getNetworkSegmentIndex(deploymentName);
     const networks = this.getNetworks(index);
     const allRequiredNetworks = _.union(networks.dynamic, networks.all.filter(net => _.startsWith(net.name, this.networkName)));
-
+    const tags = {
```

Comment:
Can we put a null check that if the opts is null/empty, it does not create the tag block.

Reply:
+1

Reply:
on second thought, tags should never be null right?

Reply:
We will have to check if opts is empty (it only has previous_manifest) ie. does not have both org and space guid. In that case we can keep tags as empty.

Reply:
opts would be null in case of non-cf cross consumption.

But I am ok with taking it up later.

Reply:
@amitmalav if in previous manifest this was empty, it will be updated post pushing this feature. So this will never be empty once this feature goes live.
@subhankarc even in case of non-cf we would have an id equivalent to org/space id? We might not have two ids, but we could still have one ID. so when we normalize it,  the tag might not be empty even for cross consumption.

Reply:
Yes, Agreed. In current implementation, yes, opts will never be empty. But I think what @subhankarc was speculating, was for generic calls to generateManifest() without space_guid and org_guid info present in opts, then tags block will have space and org guids as null. I think we can take it up later when we will be making changes to support non-cf cross-consumption.